File: javascript/jssearch/src/searchconfig.js
/*
 * searchconfig.js
 *
 * Configuration values
 */

'use strict';

const { FindConfig } = require('jsfind');
const path = require('path');

const isWin = /^win/.test(process.platform);

const HOME_NAME = isWin ? 'USERPROFILE' : 'HOME';
const HOME = process.env[HOME_NAME];

class SearchConfig extends FindConfig {
  constructor() {
    super();
    const defaultXSearchConfigDir = path.join(HOME, '.config', 'xsearch');
    const xsearchConfigDir = process.env.XSEARCH_CONFIG_DIR
      ? process.env.XSEARCH_CONFIG_DIR
      : defaultXSearchConfigDir;
    const defaultXSearchPath = path.join(HOME, 'src', 'xsearch');
    const xsearchPath = process.env.XSEARCH_PATH ? process.env.XSEARCH_PATH : defaultXSearchPath;
    const jssearchPath = path.join(xsearchPath, 'javascript', 'jssearch');
    const dataPath = path.join(jssearchPath, 'data');
    const searchOptionsPath = path.join(dataPath, 'searchoptions.json');
    const defaultSearchSettingsPath = path.join(xsearchConfigDir, 'settings.json');

    this.xsearchPath = xsearchPath;
    this.searchOptionsPath = searchOptionsPath;
    this.defaultSearchSettingsPath = defaultSearchSettingsPath;
  }
}

exports.SearchConfig = SearchConfig;

File: ruby/rbsearch/lib/rbsearch/searchconfig.rb
# frozen_string_literal: true

require 'rbfind/findconfig'

module RbSearch

  # SearchConfig - basic config info
  class SearchConfig < RbFind::FindConfig
    attr_reader :search_options_path
    attr_reader :default_search_settings_path

    def initialize
      super
      @search_options_path = File.realpath(File.join(File.dirname(__FILE__), "../../data/searchoptions.json"))
      default_xsearch_config_dir = File.join(ENV['HOME'], '.config', 'xsearch')
      xsearch_config_dir = ENV.fetch('XSEARCH_CONFIG_DIR', default_xsearch_config_dir)
      @default_search_settings_path = File.join(xsearch_config_dir, 'settings.json')
    end
  end
end
